jsettlers.logic/src/main/java/jsettlers/input/GuiInterface.java

```diff
@@ -412,10 +442,39 @@ private void selectNextOfType() {
 		}
 	}
 
+	public void setDock(ShortPoint2D requestedDockPosition) {
+		final ISelectable selected = currentSelection.getSingle();
+		if (selected instanceof Building) {
+			DockPosition dockPosition = grid.findDockPosition(requestedDockPosition);
+			if (dockPosition == null) {
+				connector.playSound(116, 1); // this dock position is not at the coast
```

Comment:
Couldn't we have some class around GuiInterface being responsible for sound resulting from user action? This could be similar to playSoundAction1() in MapObjectDrawer.java